CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+
+All sections should have headers using the appropriate LaTeX sectioning commands:
+
+| Section | Command |
+|---------|:-------:|
+| section title | `\section{}` |
+| Applications | `\subsection{}` |
+| Sample Problem | `\subsection{}` |
+| Input Format | `\subsubsection{}` |
+| Sample Input | `\subsubsection{}` |
+| Output Format | `\subsubsection{}` |
+| Sample Output | `\subsubsection{}` |
+| Lessons Learned | `\subsubsection{}` |
+
+### Code Listings
+
+Always use the `\acmlisting` command when including code, input, or output in the Hack Pack.
+The label and caption should always be set as well; both should be set to the same value.
+A code listing for a sample problem titled 'Cows Amok' would be inserted like so:
+
+`\acmlisting[label=Cows Amok, caption=Cows Amok]{./path/to/cows-amok.cpp}`
+
+Sample input for the problem would be inserted like so:
+
+`\acmlisting[label=Cows Amok Sample Input, caption=Cows Amok Sample Input]{./path/to/cows-amok.in}`
+
+Sample output for the problem would be inserted like so:
+
+`\acmlisting[label=Cows Amok Sample Output, caption=Cows Amok Sample Output]{./path/to/cows-amok.out}`
+
+### Style Changes
+
+Any changes to the style of the document must be cleared with the core Hack Pack team before implementing.
+Any submissions that change the style of document (even if not global) without prior approval will be rejected.
+
+## Condensing
+
+Each submission to the Hack Pack must also have a condensed version for inclusion in the condensed Hack Pack as well.
+This is trivial.
+
+### Content
+
+A complete, condensed section of the Hack Pack should include:
+
+- a brief description of the problem (possibly gleaned from the full description)
+- reference code (if the section is for a data structure)
+- sample solutions for each included problem
+
+### How To Condense Your Section
+
+We have defined special keywords (from a familiar language) to separate information that should be displayed in each section: `#define hackpack`, `#define hackpackpp`, and `#endif`.
+Wrap a section with the `#define hackpack` and `#endif` to include the section only in the condensed version of the Hack Pack.
+Wrap a section with the `#define hackpack` and `#endif` to include the section only in the full version of the Hack Pack.
+Any section that is not wrapped with either set of keywords will be included in both versions of the Hack Pack.
+
+## Submission
+
+Before submitting a section for inclusion in the Hack Pack, ensure that both the Hack Pack and Hack Pack++ build (using only `make`) on the McAdams lab machines _without stopping_.
+
+## Still Not Sure?
+
+We strive to make our content as clear and concise as possible, but it is always possible that we have left some content off.
+If you have any question about any part of your LaTeX, it is always a good idea to consult entries that have already made it into the Hack Pack.
+In particular, the section on INSERT SECTION HERE is an example of an excellent addition to the Hack Pack.
```

Comment:
I have been telling people `set` is the section to look for, but we should double check that all of these rules are followed there before merging.